trafficgen-infra

```diff
@@ -20,11 +20,18 @@ rate_unit="%"
 one_shot="0"
 # TODO: Auto adjust limit based on actual requirement (which is probably number of interfaces)
 mem_limit=2048
-longopts="active-devices:,devices:,cpus:,mem-limit:"
+trex_software_mode="off"
+
+longopts="active-devices:,devices:,cpus:,mem-limit:,trex-software-mode:"
 opts=$(getopt -q -o "" --longoptions "$longopts" -- "$@");
 eval set -- "$opts";
 while true; do
     case "$1" in
+	--trex-software-mode)
+	    shift
+	    trex_software_mode=${1}
+	    shift
```

Comment:
is this right?

Reply:
Yes.  If you look at the existing parameters they all follow a similar form.  What looks weird/wrong about it?